src/file_io.py

```diff
@@ -20,17 +20,22 @@ def __str__(self):
         return self.filename()
 
 
-def write_to_csv(data_list: list, outfile: File):
+def write_to_csv(data_list: list[dataclass], outfile: File):
     """Writes `data_list` to `filename` as csv"""
     print(f"dumping {len(data_list)} results to {outfile.name}")
-    headers = [f.name for f in fields(data_list[0])]
-    data_tuple = [astuple(x) for x in data_list]
-
     # Creates out path if it doesn't exist.
     if not os.path.exists(outfile.path):
         os.makedirs(outfile.path)
 
+    if len(data_list) == 0:
+        # Create an empty file
+        with open(outfile.filename(), 'w', encoding='utf-8') as _:
+            pass
+        return
+
     with open(outfile.filename(), 'w', encoding='utf-8') as out_file:
+        headers = [f.name for f in fields(data_list[0])]
+        data_tuple = [astuple(x) for x in data_list]
```

Comment:
This change should rather be a separate PR.

Also Could this be written as?:
```python
with open(outfile.filename(), 'w', encoding='utf-8') as out_file:
    if len(data_list) == 0:
        return
    headers = [f.name for f in fields(data_list[0])]
    #...
```

Reply:
For sure - will do.

Reply:
Actually, I don't like the idea of returning from within an open file. it is important the the `with open` block is completed.

Reply:
Opened PR for this specific change #7 - we can continue the discussion there.

Reply:
I really have no strong feelings about this but it seems like returning early from the `with` block would also properly close the file: https://stackoverflow.com/questions/9885217/in-python-if-i-return-inside-a-with-block-will-the-file-still-close

Reply:
If that's the case then I will surely do this!